src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -33000,6 +33000,73 @@ void Test(C? x, C? y)
                 );
         }
 
+        [Fact]
+        [WorkItem(26624, "https://github.com/dotnet/roslyn/issues/26624")]
+        public void ConditionalBranching_19()
+        {
+            CSharpCompilation c = CreateCompilation(@"
+class C
+{
+    void Test1(CL1? x1)
+    {
+        _ = x1?.P1 ?? false
+            ? x1.P1
+            : x1.P1; // 1
+
+        _ = x1?.P1 ?? true
+            ? x1.P1 // 2
+            : x1.P1;
+    }
+}
+
+class CL1
+{
+    public bool P1 { get { return true; } }
+}
+", options: WithNonNullTypesTrue());
+            c.VerifyDiagnostics(
+                // (8,15): warning CS8602: Dereference of a possibly null reference.
+                //             : x1.P1; // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x1").WithLocation(8, 15),
+                // (11,15): warning CS8602: Dereference of a possibly null reference.
+                //             ? x1.P1 // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x1").WithLocation(11, 15));
+        }
+
+        [Fact]
+        [WorkItem(26624, "https://github.com/dotnet/roslyn/issues/26624")]
+        public void ConditionalBranching_20()
+        {
+            CSharpCompilation c = CreateCompilation(@"
+class C
+{
+    void Test1(CL1? x1)
+    {
+        _ = x1?.Next?.Next?.P1 ?? false
+            ? x1.Next.Next.ToString()
```

Comment:
>x1.Next.Next [](start = 14, length = 12)

Could you test `x1.ToString()` on both branches? Maybe `? (x.Next.Next.ToString() + x.ToString()) ...`

Reply:
Done